config/clients/js/template/common.mustache

```diff
@@ -128,8 +128,47 @@ function isAxiosError(err: any): boolean {
 function randomTime(loopCount: number, minWaitInMs: number): number {
   const min = Math.ceil(2 ** loopCount * minWaitInMs);
   const max = Math.ceil(2 ** (loopCount + 1) * minWaitInMs);
-  return Math.floor(Math.random() * (max - min) + min); //The maximum is exclusive and the minimum is inclusive
+  const calculatedTime = Math.floor(Math.random() * (max - min) + min);
+  return Math.min(calculatedTime, 120000);
 }
+function parseRetryAfterHeader(headers: Record<string, string | string[] | undefined>): number | undefined {
+  const retryAfter = headers["retry-after"] || headers["Retry-After"];
+  
+  if (!retryAfter) {
+    return undefined;
+  }
+
+  const retryAfterValue = Array.isArray(retryAfter) ? retryAfter[0] : retryAfter;
+  
+  if (!retryAfterValue) {
+    return undefined;
+  }
+
+  // Try to parse as integer (seconds)
+  const secondsValue = parseInt(retryAfterValue, 10);
+  if (!isNaN(secondsValue)) {
+    const msValue = secondsValue * 1000;
+    if (msValue >= 1000 && msValue <= 1800000) {
```

Comment:
rather than these being hard-coded, now that the go-sdk changes have been merged, you can use the following variables:
- {{maxBackoffTimeInSec}}

```suggestion
    if (msValue > 0 && msValue <= {{retryHeaderMaxAllowableDurationInSec}}) {
    }
```